### PR TITLE
fix(dashpay): update image permissions for Android 13 and fix camera

### PR DESCRIPTION
**Pull request by HashEngineering**
## Issue being fixed or feature implemented

* Support Android 13 (images)
* Refactor to use ActivityResultLauncher
## How Has This Been Tested?

- [ ] QA (Mobile Team)

## Checklist:

- [x] I have performed a self-review of my own code and added comments where necessary
- [ ] I have added or updated relevant unit/integration/functional/e2e tests

### CODE REVIEW

File: wallet/src/de/schildbach/wallet/ui/dashpay/CropImageActivity.kt [hunk 1]
```diff
@@ -73,6 +73,7 @@ class CropImageActivity : LockScreenActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         binding = ActivityCropImageBinding.inflate(layoutInflater)
+        setContentView(binding.root)
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/dashpay/CropImageActivity.kt:76]: This was left out from the previous refactoring.

File: wallet/src/de/schildbach/wallet/ui/EditProfileActivity.kt [hunk 2]
```diff
@@ -207,6 +209,81 @@ class EditProfileActivity : LockScreenActivity() {
...
+                // Handle the case where permission is denied
+            }
+        }
+
+        takePictureLauncher = registerForActivityResult(ActivityResultContracts.TakePicture()) { isSuccess: Boolean ->
+            if (isSuccess) {
+                editProfileViewModel.onTmpPictureReadyForEditEvent.postValue(editProfileViewModel.tmpPictureFile)
+            }
+            turnOnAutoLogout()
+        }
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/EditProfileActivity.kt:226]: This refactoring also solved the problem of failure of the camera on Android 12 and 14 emulators, though the old code did work on a real Android 13 device.

File: wallet/src/de/schildbach/wallet/ui/dashpay/HistoryHeaderAdapter.kt [hunk 3]
```diff
@@ -199,6 +199,6 @@ class HistoryHeaderAdapter(
 
     private fun shouldShowJoinDashPay(canJoin: Boolean): Boolean {
         val hideJoinDashPay = preferences.getBoolean(PREFS_KEY_HIDE_JOIN_DASHPAY_CARD, false)
-        return blockchainIdentityData == null && canJoin && !hideJoinDashPay
+        return blockchainIdentityData?.creationState == BlockchainIdentityData.CreationState.NONE && canJoin && !hideJoinDashPay
```
Comment (HashEngineering) [inline wallet/src/de/schildbach/wallet/ui/dashpay/HistoryHeaderAdapter.kt:202]: This is what will show the Join DashPay on the home screen.  Changes regarding `blockchainIdentityData` (perhaps migration from a DB table to DataStore) broke this and always returned false.

**Review by Syn-McJ — APPROVED** [review]
Looks good

**Pull request merged**